doc/banner/banner_hld.md

```diff
@@ -0,0 +1,263 @@
+# Banner messages HLD #
+
+## Table of contents
+- [Revision](#revision)
+- [About this manual](#about-this-manual)
+- [Scope](#scope)
+- [1 Introduction](#1-introduction)
+  - [1.1 Feature overview](#11-feature-overview)
+  - [1.2 Requirements](#12-requirements)
+- [2 Design](#2-design)
+  - [2.1 Overview](#21-overview)
+  - [2.2 Flows](#25-flows)
+  - [2.3 CLI](#26-cli)
+      - [2.3.1 Config command group](#231-config-command-group)
+      - [2.3.2 Show command](#232-show-command)
+  - [2.4 YANG model](#24-yang-model)
+- [3 Test plan](#3-test-plan)
+  - [3.1 Unit tests via VS](#31-unit-tests-via-vs)
+
+
+# Revision
+| Rev | Date       | Author              | Description     |
+|:---:|:----------:|:-------------------:|:----------------|
+| 0.1 | 01/02/2023 | Sviatoslav Boichuk  | Initial version |
+
+
+# About this manual
+This document provides a high-level information for the SONiC Banner feature and Banner CLI. It describes high-level behavior, internal definition, design of the commands, syntax, and output definition.
+
+# Scope
+The scope of this document is to cover definition, design and implementation of SONiC Banner feature and Banner CLI.
+The document covers the next CLI:
+1.	Commands to configure banner messages
+2.	Command to display banner settings
+
+<!-- omit in toc -->
+## Abbreviations
+| Term  | Meaning                                   |
+|:------|:------------------------------------------|
+| SONiC | Software for Open Networking in the Cloud |
+| MOTD  | Message of the day                        |
+| DB    | Database                                  |
+| CLI   | Сommand-line Interface                    |
+| YANG  | Yet Another Next Generation               |
+
+<!-- omit in toc -->
+## List of figures
+- [Figure 1: Banner system chart diagram](#figure-1-banner-system-chart-diagram)
+- [Figure 2: Banner init flow](#figure-2-banner-init-flow)
+- [Figure 3: Banner config flow](#figure-3-banner-config-flow)
+- [Figure 4: Banner show configuration](#figure-4-banner-show-configuration)
+
+# 1 Introduction
+
+## 1.1 Feature overview
+The SONiC maintains several messages for communication with users. These messages are associated with login and logout processes.
+
+There are few banners message types:
+|  Type         |  Description                                                                    |
+| :-----------: | :-----------------------------------------------------------------------------: |
+|  login        | Display a banner to the users connected locally or remotely before login prompt |
+|  motd         | Display a banner to the users after login prompt                                |
+|  logout       | Display a logout banner to the users connected locally or remotely              |
```

Comment:
1. Please explain the purpose or use case of logout banner requirement.
2. Mention the Linux banner files (/etc/banner, /etc/motd, etc) where the banners are maintained for completeness

Reply:
1. Purpose to have logout message is - full interface (login-motd-logout)
2. Updated HLD and mentioned Linux files

Reply:
Sorry, still not convinced about the use case for logout banner. Linux does not support logout banner natively, seems to be something new being brought as SONiC specific. I would recommend to add logout banner, if valid use case is available.

Reply:
Logout message supported by bash (bash.bash_logout file). The logout message will be written from ConfDB to new file: /etc/logout_message. And "bash.bash_logout" will be modified to read logout message from "/etc/logout_message" and display it (via 'echo').  Some vendors, for example Mellanox (https://docs.nvidia.com/networking/display/onyxv3102002/ui+commands#src-80577389_UICommands-Banner) support logout banner, so it might be useful for users.